### PR TITLE
adds mergesort

### PR DESCRIPTION
**Pull request by lizzkats**

### CODE REVIEW

File: src/mergesort.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+export default function mergeSort(array) {
+  if (!Array.isArray(array)) return "We can only sort arrays"
+  
+  if(array.length < 2) return array
+
+  var middle = Math.floor(array.length / 2)
+  var left = array.slice(0, middle)
+  var right = array.slice(middle, array.length)
+
+  return stitch(mergeSort(left), mergeSort(right))
```
Comment (jrobcodes) [inline src/mergesort.js:10]: By convention, this function is usually called `merge`, but I like the name nonetheless.  :P

File: src/mergesort.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+function stitch(left, right){
```
Comment (jrobcodes) [inline src/mergesort.js:13]: I like how your implementation leverages the builtin `shift` function.  I encourage you to consider how you would implement this if no `shift` were available (which is typical of languages like Java and C++).

File: test/mergesort_test.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { expect } from 'chai'
+import mergeSort from '../src/mergeSort.js'
+
+describe('mergeSort()', function(){
```
Comment (jrobcodes) [inline test/mergesort_test.js:4]: Consider adding tests for empty arrays, and arrays that are already in sorted order, to make sure you're covering all possible cases.  You should also probably have tests for arrays with multiple types?
